client/pages/spotify-auth-redirect.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+
+export default class SpotifyAuthRedirect extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isAuthorizing: true,
+      signedIn: false,
+      username: ''
+    };
+  }
+
+  componentDidMount() {
+    const params = new URLSearchParams(window.location.search);
+    const authCode = params.get('code');
+
+    const postBody = {
+      client_id: process.env.REACT_APP_SPOTIFY_CLIENT_ID,
+      grant_type: 'authorization_code',
+      code: authCode,
+      redirect_uri: 'http://localhost:3000/callback',
+      code_verifier: sessionStorage.getItem('spotify-code-verifier')
+    };
+
+    const checkState = params.get('state');
+    if (checkState !== sessionStorage.getItem('spotify-state')) {
+      return undefined;
+    } else {
+      const init = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded;'
+        },
+        body: new URLSearchParams(postBody)
+      };
+      fetch('https://accounts.spotify.com/api/token', init)
+        .then(result => {
+          return result.json();
+        })
+        .then(entry => {
+          const accessInfo = entry;
+          const accessToken = accessInfo.access_token;
+          const authHeader = {
+            headers: { Authorization: 'Bearer ' + accessToken }
+          };
+          fetch('https://api.spotify.com/v1/me', authHeader)
```

Comment:
If you add a `return` in front of `fetch()`, you can remove the `.catch()` on lines 55-57